File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table  = {}
        for i, num in enumerate(nums):
            left = target - num
            if num in table:
                return [table[num], i]
            else:
                table[left] = i
            

File: 1-two-sum/two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    final Map<int, int> vals = {};
    for (int i = 0; i < nums.length; i++) {
      final int val = nums[i];
      if (vals.containsKey(target - val)) {
        return [i, vals[target - val]!];
      } else {
        vals[val] = i;
      }
    }
    return [];
  }
}